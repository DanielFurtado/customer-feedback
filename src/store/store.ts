import { configureStore } from '@reduxjs/toolkit';
import ratingReducer from './rating/rating.reducer';

export const store = configureStore({
  reducer: {
    feedbackList: ratingReducer
  }
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
