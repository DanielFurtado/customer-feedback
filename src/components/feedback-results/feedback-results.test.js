import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import FeedbackResults from './feedback-results';
import {BrowserRouter as Router} from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../store/store';

describe(('Feedback Results'), () => {
  it('renders', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <FeedbackResults />
        </Router>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});