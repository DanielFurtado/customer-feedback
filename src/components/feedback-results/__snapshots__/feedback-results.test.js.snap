// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Results renders 1`] = `
<div
  className="sc-eDWCr himBEe"
>
  <div
    className="sc-bqWxrE gGrhSY"
  >
    <h2>
      Feedback Results
    </h2>
    <button
      className="sc-fnGiBr jDaNuL"
      onClick={[Function]}
      type="button"
    >
      Write a review
    </button>
  </div>
  <section>
    <div
      className="sc-gKPRtg hyPmrB"
    >
      <div
        className="sc-iBYQkv bctHwM"
      >
        <svg>
          info-icon.svg
        </svg>
      </div>
      No ratings yet!
    </div>
  </section>
  <h3>
    Latest Comments
  </h3>
  <section>
    <div
      className="sc-gKPRtg hyPmrB"
    >
      <div
        className="sc-iBYQkv bctHwM"
      >
        <svg>
          info-icon.svg
        </svg>
      </div>
      Be the first to submit your feedback.
    </div>
  </section>
</div>
`;
